doc/source/basics.rst

```diff
@@ -793,6 +793,79 @@ This is equivalent to the following
    result
    result.loc[:,:,'ItemA']
 
+.. _basics.assigning:
+
+Assigning New Columns
+---------------------
+
+.. versionadded:: 0.16.0
+
+Inspired by `dplyr's
+<http://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html#mutate>`__
+``mutate`` verb, DataFrame has an :meth:`~pandas.DataFrame.assign`
+method that allows you to easily create new columns that are potentially
+derived from existing columns. 
+
+.. ipython:: python
+
+   iris = read_csv('data/iris.data')
+   iris.head()
+
+   iris.assign(sepal_ratio=iris['SepalWidth'] / iris['SepalLength']).head()
+
+Above was an example of inserting a precomputed value. We can also pass in
+a function of one argument to be evalutated on the DataFrame being assigned to.
+
+.. ipython:: python
+
+   iris.assign(sepal_ratio = lambda x: x['SepalWidth'] /
+               x['SepalLength']).head()
+
+The power of ``assign`` comes when used in chains of operations. For example,
+we can limit the DataFrame to just those observations with a Sepal Length
+greater than 5, calculate the ratio, and plot:
+
+.. ipython:: python
+
+   (iris.query('SepalLength > 5')
+        .assign(SepalRatio=lambda x: x.SepalWidth / x.SepalLength,
+                PetalRatio=lambda x: x.PetalWidth / x.PetalLength)
+        .plot(kind='scatter', x='SepalRatio', y='PetalRatio')
+    )
+
+Since a function is passed in, the function is computed on the DataFrame
+being assigned to. Importantly, this is the DataFrame that's been filtered
+to those rows with sepal length greater than 5.
+
+The function signature for ``assign`` is simply ``**kwargs``. The keys
+are the column names for the new fields, and the values are either a value
+to be inserted (for example, a ``Series`` or NumPy array), or a function
+of one argument to be called on the ``DataFrame``. The new values are inserted,
+and the entire DataFrame (with all original and new columns) is returned.
+
+.. warning::
+
+  Since the function signature of ``assign`` is ``**kwargs**``, a dictionary,
+  the order of the columns in the result DataFrame cannot be guarunteed.
+
+  All expressions are computed first, and then assigned. So you can't refer
+  to another column being assigned in the same call to ``assign``. For example:
+
+   .. code-block:: python
+
+         >>># Don't do this
```

Comment:
I think we usually use IPython style `In [1]` prompts rather than `>>>`.

Also, there's a nice IPython verbatim sphinx directive that will number lines properly even if they aren't evaluated, e.g., write something like:
```
.. ipython::
    :verbatim:

    In [1]: df['not_found']
    Out[1]: KeyError
```

Probably best to try building the docs to make sure it looks right.